platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/tipset/ThrottledTipsetEventCreator.java

```diff
@@ -0,0 +1,165 @@
+
+import com.swirlds.base.time.Time;
+import com.swirlds.common.context.PlatformContext;
+import com.swirlds.common.system.EventCreationRuleResponse;
+import com.swirlds.common.system.status.PlatformStatus;
+import com.swirlds.common.threading.framework.QueueThread;
+import com.swirlds.common.utility.throttle.RateLimiter;
+import com.swirlds.platform.StartUpEventFrozenManager;
+import com.swirlds.platform.event.EventIntakeTask;
+import com.swirlds.platform.event.GossipEvent;
+import com.swirlds.platform.eventhandling.EventTransactionPool;
+import com.swirlds.platform.internal.EventImpl;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.util.Objects;
+import java.util.function.Supplier;
+
+/**
+ * Responsible for creating new events using the tipset algorithm. Similar behavior as a {@link TipsetEventCreatorImpl},
+ * but will sometimes choose not to create events for reasons other than the tipset algorithm.
+ */
+public class ThrottledTipsetEventCreator implements TipsetEventCreator {
+
+    /**
+     * Prevent new events from being created if the event intake queue ever meets or exceeds this size.
+     */
+    private final int eventIntakeThrottle;
+
+    private final RateLimiter rateLimiter;
+    private final EventTransactionPool transactionPool;
+    private final QueueThread<EventIntakeTask> eventIntakeQueue;
+    private final Supplier<PlatformStatus> platformStatusSupplier;
+    private final StartUpEventFrozenManager startUpEventFrozenManager;
+
+    private final TipsetEventCreator baseCreator;
+
+    /**
+     * Constructor.
+     *
+     * @param platformContext           the platform's context
+     * @param time                      provides the wall clock time
+     * @param transactionPool           provides transactions to be added to new events
+     * @param eventIntakeQueue          the queue to which new events should be added
+     * @param platformStatusSupplier    provides the current platform status
+     * @param startUpEventFrozenManager prevents event creation when the platform has just started up
+     * @param baseCreator               the base tipset event creator wrapped by this object
+     */
+    public ThrottledTipsetEventCreator(
+            @NonNull final PlatformContext platformContext,
+            @NonNull final Time time,
+            @NonNull final EventTransactionPool transactionPool,
+            @NonNull final QueueThread<EventIntakeTask> eventIntakeQueue,
+            @NonNull final Supplier<PlatformStatus> platformStatusSupplier,
+            @NonNull final StartUpEventFrozenManager startUpEventFrozenManager,
+            @NonNull final TipsetEventCreator baseCreator) {
+
+        eventIntakeThrottle = platformContext
+                .getConfiguration()
+                .getConfigData(EventCreationConfig.class)
+                .eventIntakeThrottle();
+
+        final EventCreationConfig eventCreationConfig =
```

Comment:
You could switch this statement with the last and use the result of this one to get the other

Reply:
done